### PR TITLE
rewrite the function `ensure_any_column_reference_is_unambiguous`

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi <13716567376yh@gmail.com>

# Which issue does this PR close?

None.

# Rationale for this change

As the columns referenced in `expr` are always fewer than the columns in `schemas`, we don't need to include all the columns in our hash map.

# What changes are included in this PR?

# Are these changes tested?
Pass current tests.

# Are there any user-facing changes?
No.

### CODE REVIEW

**Comment by HaoYang670** [comment]
Could you please help to review @ygf11 ?

**Comment by ygf11** [comment]
Make sense to me. Thank you @HaoYang670.

File: datafusion/sql/src/planner.rs [hunk 1]
```diff
@@ -3251,43 +3251,44 @@ fn ensure_any_column_reference_is_unambiguous(
...
+    // find the name of the column existing in multi schemas.
+    let ambiguous_col_name = schemas
         .iter()
         .flat_map(|schema| schema.fields())
-        .for_each(|field| {
-            column_count_map
-                .entry(field.name())
-                .and_modify(|v| *v += 1)
-                .or_insert(1usize);
+        .map(|field| field.name())
```
Comment (jackwener) [inline datafusion/sql/src/planner.rs:3270]: .flat_map(|schema| schema.fields())
        .map(|field| field.name())

->

        .flat_map(|schema| schema.fields().name())

Comment (jackwener) [reply datafusion/sql/src/planner.rs:3270]: > Op, sorry we could not do this as the return type of `fields()` is `Vec<DFField>`.

😂my mistake

**Comment by alamb** [comment]
Thank you @HaoYang670  and @ygf11  and @jackwener  for the review

**Pull request merged**